### DISCUSSION TITLE
I get an error ". . . could not assemble any primary key columns . . . " when I don't believe I should

### DISCUSSION THREAD

**Discussion by hbuhrmann** [sqlalchemy/sqlalchemy#9577] [Usage Questions] [answered]
I am using sqlalchemy 2.07

I am trying to use reflection to create my table classes, as follows

```
metadata_obj = MetaData()

db_session = Session(engine)

class Base(DeclarativeBase):
    pass

Base.metadata.reflect(engine)

class EventPivot(Base):
    __tablename__ = Base.metadata.tables['nm_event_pivot']
```

However, when I run this code, I get the following error:

_sqlalchemy.exc.ArgumentError: Mapper Mapper[EventPivot(nm_event_pivot)] could not assemble any primary key columns for mapped table 'nm_event_pivot'_

Which really doesn't make sense, as the primary key has definitely been identified. 

In pycharm debugger:
`Base.metadata.tables['nm_event_pivot'].primary_key `

returns the following :

`PrimaryKeyConstraint(Column('auto_id', INTEGER(), table=<nm_event_pivot>, primary_key=True, nullable=False, server_default=DefaultClause(<sqlalchemy.sql.elements.TextClause object at 0x000002ADBC0E6980>, for_update=False)))`

I have to use reflection, as the columns in the table continuously change so I am not quite sure how to fix this. Any ideas would be appreciated

**Comment by zzzeek** [comment, marked as answer]
for guidance on mapping with reflection see the section https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#mapping-declaratively-with-reflected-tables .

in this case, ``__tablename__`` refers to the string name of a table, not an actual `Table` object.  you want to use ``__table__``:

```py
class EventPivot(Base):
    __table__ = Base.metadata.tables['nm_event_pivot']
```

**Comment by hbuhrmann** [comment]
Many thanks Mike. I have been staring at the code for too long . . .
